### PR TITLE
updated README.

### PR DESCRIPTION
**Pull request by Mackey22**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+The entry to this package is the key word :code:`hca`. Merely typing :code:`hca` will list all commands that are possible in a manner similar to the above. 
+
+Each of the above commands has its own associated optional or required arguments. To see these, type :code:`hca <command> -h`. These arguments are listed in a style common to most argparse parsers. For instance: 
+
+:code:`hca put-bundles [-h] [--version VERSION] --replica REPLICA --files INDEXED/NAME/UUID/VERSION [INDEXED/NAME/UUID/VERSION ...] --creator-uid CREATOR_UID uuid`
+
+Here is a list of what each type means. 
+
+- Positional arguments: These don't have a flag in front of them. If they are surrounded by brackets, they are optional.
+- Optional agruments: Note that the term optional in this sense doesn't mean that the argument is not required. It means that the argument is identified by a flag. If the argument and the flag (A word with "--" in front of it). are surrounded in brackets, this input is optional. If an arument is repeated multiple times with the later arguments surrounded in brackets, it means the parser accepts a list of these arguments.
```
Comment (ttung) [inline README.rst]: *arguments

File: README.rst [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+
+Here is a list of what each type means. 
+
+- Positional arguments: These don't have a flag in front of them. If they are surrounded by brackets, they are optional.
+- Optional agruments: Note that the term optional in this sense doesn't mean that the argument is not required. It means that the argument is identified by a flag. If the argument and the flag (A word with "--" in front of it). are surrounded in brackets, this input is optional. If an arument is repeated multiple times with the later arguments surrounded in brackets, it means the parser accepts a list of these arguments. 
+- Objects: Because the REST API sometimes consumes lists of objects, it is important to be able to pass these into the command line interface. To pass in an object, it is a number of arguments separated by slashes, as seen in the example above. An example input to this could be :code:`True/Bob/3806d74a-6ab5-4a6d-ba00-667ea858c7b5/2017-06-30T19:33:38+00:00`. If an argument within an object is optional and you don't have an input for it, you can replace it's supposed place with :code:`None`.
```
Comment (ttung) [inline README.rst]: I'd avoid True since it's actually a python keyword.

Comment (Mackey22) [inline README.rst]: It has to be True or False because I'm parsing it as a boolean.

File: README.rst (hunk 1 above)
Comment (Mackey22) [inline README.rst]: Changing

File: README.rst (hunk 2 above)
Comment (kislyuk) [inline README.rst]: Consider loading it at JSON instead

File: .gitignore [hunk 3]
```diff
@@ -2,6 +2,15 @@
...
 # - No leading slash means the pattern matches at any depth.
 
+# Sublime files
+*~
+
+# Random files
+get-pip.py
+setup.py-e
+test/play.py
+v/
```
Comment (kislyuk) [inline .gitignore]: I think the only one that really belongs in .gitignore is `setup.py-e`. The others are not generic build byproducts or anything like that, they are changes that you named, so they should be in your personal .gitignore (you can put one in your home directory I believe).

File: Changes.rst [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+Command line interface that interacts with the Human Cell Atlas data
+store REST API.
+
+Current functionality: Retrieve or register a data file. Retrieve or
+register a data bundle (collection of data files). Upload full directory
+to cloud, register given files, and collect them into a bundle. Download
+a bundle or file from HCA data store to local.
+
+Changes for v1.0.0 (2017-06-26)
+===============================
```
Comment (kislyuk) [inline Changes.rst]: FYI, the release script (in common.mk) uses the Git history to automatically generate a changelog for you. It then prepends the changelog to this file under the heading that corresponds to the new version (auto-generated). At the time of the release, you can edit the changelog and put this text in there, but the text that you've added here won't actually show up in the release notes on PyPI or Github.

**Comment by Mackey22** [comment]
PR approved by @ttung merging into Master.

File: README.rst [hunk 5]
```diff
@@ -1,5 +1,38 @@
 HCA DSS CLI
 ===========
+This repository contains a prototype for interacting with the replicated data storage system (aka the "blue box") of the Human Cell Atlas.
```
Comment (kislyuk) [inline README.rst]: Please wrap lines at 120.

Comment (kislyuk) [inline README.rst]: In my personal opinion, we should avoid using the "blue/green/red box" terminology in formal documentation. Feel free to push back though.

File: README.rst [hunk 6]
```diff
@@ -1,5 +1,38 @@
...
+This prototype uses a local version of the blue box API spec (using Swagger) to generate a command line client that will make requests to the blue box. Currently, the commands listed are: 
+- upload        Upload a file or directory to the cloud, register each of the files, and bundle them together.
+- download      Download a full bundle or file to local.
+- get-search    Returns a list of bundles matching the given simple criteria.
+- post-search   Accepts Elasticsearch JSON query and returns matching bundle identifiers.
+- get-files     Returns a list of files matching given criteria.
+- head-files    Get the associated metadata from a file with a given uuid.
+- put-files     Create a new file with a given UUID. The file content is passed in through a cloud URL. The file on the cloud provider must have metadata set reflecting the file checksums and the file content type.
+- get-bundles   Returns a list of bundles matching given criteria.
+- put-bundles   Create a new bundle with a given UUID. The list of files UUID+versions to be included must be provided.
```
Comment (kislyuk) [inline README.rst]: Consider making sure these docstrings are present in the data-store swagger YAML, and (for basic wrapper commands, not upload/download) in `hca --help`. Then consider replacing this block of text with something like "Please run `hca --help` to see an overview of available commands".

File: README.rst [hunk 7]
```diff
@@ -1,5 +1,38 @@
...
+
+Installation
+------------
+To install this package, run :code:`pip install hca-cli`. This will automatically hook up with the api endpoint as defined in the package published last. 
+
+To use the command line interface with a local or test blue box, open <directory_holding_hca_module>/hca/api_spec.json. Change :code:`host` to the host you want (if you're running on a local blue box, this will likely be :code:`localhost:5000`) and the first argument of :code:`schemes` should be the scheme you want (:code:`http` if running locally, :code:`https` otherwise).
+
+Usage
+-----
+The entry to this package is the key word :code:`hca`. Merely typing :code:`hca` will list all commands that are possible in a manner similar to the above.
```
Comment (kislyuk) [inline README.rst]: This wording seems a bit awkward, maybe rephrase as "The command-line utility `hca` is the entry point to the CLI provided by this package".

File: setup.cfg [hunk 8]
```diff
@@ -2,4 +2,4 @@
 universal=1
 [flake8]
 max-line-length=120
-ignore: E301, E302, E401, E261, E265, E226, F401, E501
+ignore: E301, E302, E305, E401, E402, E261, E265, E226, F401, E501
```
Comment (kislyuk) [inline setup.cfg]: Instead of adding ignores here, please mark offending lines with `# noqa`.

File: .gitignore (hunk 3 above)
Comment (Mackey22) [inline .gitignore]: Should I not include the sublime files either?

File: Changes.rst (hunk 4 above)
Comment (Mackey22) [inline Changes.rst]: Yeah, this was from when I was trying to publish the package to pip myself but it wasn't going through. Should I remove this stuff on master, then?

File: README.rst (hunk 5 above)
Comment (Mackey22) [inline README.rst]: 👍

Comment (Mackey22) [inline README.rst]: Makes sense. Will change to `DSS` in all other references

File: README.rst (hunk 6 above)
Comment (Mackey22) [inline README.rst]: 👍

File: README.rst (hunk 7 above)
Comment (Mackey22) [inline README.rst]: 👍

File: .gitignore (hunk 3 above)
Comment (kislyuk) [inline .gitignore]: Sublime files are fine, I think, like any files with a common pattern unique to an application used by developers.

File: Changes.rst (hunk 4 above)
Comment (kislyuk) [inline Changes.rst]: Oh, no problem, sorry I forgot about that. I'll fix up the changelog when I cut the next release, then.
